python_modules/dagster/dagster/_core/definitions/asset_condition.py

```diff
@@ -175,65 +217,54 @@ def with_run_ids(self, run_ids: AbstractSet[str]) -> "AssetConditionEvaluationWi
         return AssetConditionEvaluationWithRunIds(evaluation=self, run_ids=frozenset(run_ids))
 
 
-class AssetConditionEvaluationResult(NamedTuple):
-    """Return value for the evaluate method of an AssetCondition."""
+@whitelist_for_serdes
+class AssetConditionEvaluationState(NamedTuple):
+    """Incremental state calculated during the evaluation of an AssetCondition. This may be used
+    on the subsequent evaluation to make the computation more efficient.
+    """
 
-    condition: "AssetCondition"
     evaluation: AssetConditionEvaluation
-    extra_values_by_unique_id: Mapping[str, PackableValue]
+    evaluation_timestamp: Optional[float]
+
+    max_storage_id: Optional[int]
+    extra_state_by_unique_id: Mapping[str, PackableValue]
+
+    @property
+    def asset_key(self) -> AssetKey:
+        return self.evaluation.asset_key
 
     @property
     def true_subset(self) -> AssetSubset:
         return self.evaluation.true_subset
 
-    @staticmethod
-    def create_from_children(
-        context: "AssetConditionEvaluationContext",
-        true_subset: AssetSubset,
-        child_results: Sequence["AssetConditionEvaluationResult"],
-    ) -> "AssetConditionEvaluationResult":
-        """Returns a new AssetConditionEvaluationResult from the given child results."""
-        return AssetConditionEvaluationResult(
-            condition=context.condition,
-            evaluation=AssetConditionEvaluation(
-                context.condition.snapshot,
-                true_subset=true_subset,
-                candidate_subset=context.candidate_subset,
-                start_timestamp=context.start_timestamp,
-                end_timestamp=pendulum.now("UTC").timestamp(),
-                subsets_with_metadata=[],
-                child_evaluations=[child_result.evaluation for child_result in child_results],
-            ),
-            extra_values_by_unique_id=dict(
-                item
-                for child_result in child_results
-                for item in child_result.extra_values_by_unique_id.items()
-            ),
-        )
-
     @staticmethod
     def create(
```

Comment:
Similar to my feedback on an earlier PR, worth adding a short comment explaining why there's a `create` static method instead of constructor.